File: Spring_Backend/src/main/java/polsl/tab/skiresort/api/passes/response/PassResponse.java
package polsl.tab.skiresort.api.passes.response;

import polsl.tab.skiresort.model.Pass;

import java.sql.Date;
import java.sql.Timestamp;

public class PassResponse {

    private final Integer id;

    private final Float unitPrice;

    private final Timestamp startDate;

    private final Timestamp endDate;

    private final String firstName;

    private final String lastName;

    private final Date birthDate;

    private final Integer usesTotal;

    private final Integer usesLeft;

    private final Boolean blocked;

    public PassResponse(Pass pass) {
        this.id = pass.getIdPass();
        this.unitPrice = pass.getUnitPrice();
        this.startDate = pass.getStartDate();
        this.endDate = pass.getEndDate();
        this.firstName = pass.getFirstName();
        this.lastName = pass.getLastName();
        this.birthDate = pass.getBirthDate();
        this.usesTotal = pass.getUsesTotal();
        this.usesLeft = pass.getUsesLeft();
        this.blocked = pass.getBlocked();
    }

    public Integer getId() {
        return id;
    }

    public Float getUnitPrice() {
        return unitPrice;
    }

    public Timestamp getStartDate() {
        return startDate;
    }

    public Timestamp getEndDate() {
        return endDate;
    }

    public String getFirstName() {
        return firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public Date getBirthDate() {
        return birthDate;
    }

    public Integer getUsesTotal() {
        return usesTotal;
    }

    public Integer getUsesLeft() {
        return usesLeft;
    }

    public Boolean getBlocked() {
        return blocked;
    }
}

File: Angular_Frontend/skiresort/src/app/classes/pass-response.ts
export class PassResponse {
  id: number;
  unitPrice: number;
  firstName: string;
  lastName: string;
  startDate: any;
  endDate: any;
  birthDate: string;
  usesTotal: number;
  usesLeft: number;
  blocked: boolean;
  isActive: boolean;


  constructor(id: number, unitPrice: number, firstName: string, lastName: string, startDate: any,
              endDate: any, birthDate: string, usesTotal: number, usesLeft: number, blocked: boolean) {
    this.id = id;
    this.unitPrice = unitPrice;
    this.firstName = firstName;
    this.lastName = lastName;
    this.startDate = startDate;
    this.endDate = endDate;
    this.birthDate = birthDate;
    this.usesTotal = usesTotal;
    this.usesLeft = usesLeft;
    this.blocked = blocked;
    if (startDate || endDate) {
      this.isActive = (new Date() < new Date(endDate)) ? true : false;
    } else {
      this.isActive = usesLeft > 0 ? true : false;
    }
  }

  checkIfActive() {

  }
}
